unified_planning/model/problem.py

```diff
@@ -555,15 +566,38 @@ def add_goal(
         if goal_exp != self._env.expression_manager.TRUE():
             self._goals.append(goal_exp)
 
+    def add_trajectory_constraint(
+        self, constraint: "up.model.expression.BoolExpression"
+    ):
+        """
+        Adds the given `trajectory_constraint` to the `Problem`;
+        a trajectory_constraint is an expression defined as:
+            Always, Sometime, At-Most-Once, Sometime-Before, Sometime-After or
+            defined with quantifiers.
+
+        :param trajectory_constraint: The expression added to the `Problem`.
+        """
+        (constraint_exp,) = self._env.expression_manager.auto_promote(constraint)
```

Comment:
Here we could check that the constraint is syntactically correct (i.e. contains exactly one temporal operator at the top or a forall of temporal operators and no nesting)